browserpass.go

```diff
@@ -47,6 +49,18 @@ type msg struct {
 	Entry    string `json:"entry"`
 }
 
+func SendError(err error, stdout io.Writer) error {
```

Comment:
My special THANK YOU for this!

I'm going to separately test this more, but what is your opinion on this as a fix for #164? I personally want to see errors logged and reported back to the extension, this code already sends the message back and logs it to stdout, I would only add logging to a file too (not necessarily in this PR). Anything you would additionally want for better error reporting?

Reply:
> My special THANK YOU for this!

You're most welcome :-).

I think this goes partway to addressing #164, but I don't consider it a full fix.

I added it because I needed a way to get error info out of the host app while I was debugging it, and figured I'd do so in a way that was more generally useful. So it helps - but I don't think that #164 should be closed because of it. I suspect there's still a fair bit of error information that may not be caught.

Reply:
Besides, my way of doing it here is a little messy ;-). Anything that's returned as a raw string is considered an error. In the longer term, I'd like to see the return message standardised properly, and include a status code. But I couldn't do that without breaking backwards-compatibility, hence the string thing.

Reply:
I had that idea too, but then I saw your approach with strings and I thought that perhaps this is actually fine, we can log more details in the log file (detailed messages, error code, etc), and have the extension receive just a string error. That way we don't break compatibility, and it's easy to add to ISSUE_TEMPLATE an ask to print the last 100 lines from `/var/log/browserpass.log` 🙂

Reply:
It's not really fine in the longer term though, because at the moment we can't change much functionality-wise without breaking stuff (and besides, co-opting the entire `string` datatype for errors isn't very nice, because nobody can use it for anything else). Switching to a consistent and extensible message format will allow much more scope to add functionality without requiring a host app update every time we do something significant.

I don't think this PR is the right place for such a change, but IMO it definitely needs to happen at some point.

Reply:
Agreed, and I will update the ticket with your comment :)